diff -up kdelibs-3.5.10/kate/part/katebuffer.cpp~ kdelibs-3.5.10/kate/part/katebuffer.cpp
--- kdelibs-3.5.10/kate/part/katebuffer.cpp~	2017-02-21 09:44:34.000000000 +0100
+++ kdelibs-3.5.10/kate/part/katebuffer.cpp	2017-02-21 09:48:20.132333343 +0100
@@ -919,7 +919,7 @@ void KateBuffer::updatePreviousNotEmptyL
   } while (textLine->firstChar()==-1);
   kdDebug(13020)<<"updatePreviousNotEmptyLine: updating line:"<<(blk->startLine()+current_line)<<endl;
   QMemArray<uint> foldingList=textLine->foldingListArray();
-  while ( (foldingList.size()>0)  && ( std::abs(foldingList[foldingList.size()-2])==1)) {
+  while ( (foldingList.size()>0)  && ( std::abs((long)foldingList[foldingList.size()-2])==1)) {
     foldingList.resize(foldingList.size()-2,QGArray::SpeedOptim);
   }
   addIndentBasedFoldingInformation(foldingList,addindent,deindent);
