--- kdelibs-3.5.10/kdeprint/cups/cupsdconf2/cups-util.c.orig	2014-11-11 21:07:48.000000000 +0100
+++ kdelibs-3.5.10/kdeprint/cups/cupsdconf2/cups-util.c	2014-11-11 22:01:12.743376515 +0100
@@ -167,7 +167,7 @@
 	*/
 
 	snprintf(plain, sizeof(plain), "%s:%s", cupsUser(), pwdstring);
-#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+#if (CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2) || (CUPS_VERSION_MAJOR > 1)
        httpEncode64_2(encode, sizeof(encode), plain, sizeof(plain));
 #else
        httpEncode64(encode, plain);
@@ -430,7 +430,7 @@
 	*/
 
 	snprintf(plain, sizeof(plain), "%s:%s", cupsUser(), pwdstring);
-#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+#if (CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2) || (CUPS_VERSION_MAJOR > 1)
        httpEncode64_2(encode, sizeof(encode), plain, sizeof(plain));
 #else
        httpEncode64(encode, plain);
@@ -512,7 +512,7 @@
   * See if we are accessing localhost...
 	the struct has changed in newer versions - PiggZ (adam@piggz.co.uk)
   */
-#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+#if ( CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2 ) || (CUPS_VERSION_MAJOR > 1)
         if (!httpAddrLocalhost(http))
 #else
 	if (ntohl(*(int*)&http->hostaddr.sin_addr) != 0x7f000001 &&
--- kdelibs-3.5.10/kdeprint/cups/ipprequest.cpp~	2014-11-11 21:07:48.000000000 +0100
+++ kdelibs-3.5.10/kdeprint/cups/ipprequest.cpp	2014-11-11 22:02:13.353376077 +0100
@@ -751,7 +751,7 @@
 	cupsFreeOptions(n, options);
 
 	// find an remove that annoying "document-format" attribute
-#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+#if (CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2) || (CUPS_VERSION_MAJOR > 1)
     ipp_attribute_t *attr = ippFindAttribute(request_, "document-format", IPP_TAG_NAME);
     ippDeleteAttribute(request_, attr);
 #else
